### ISSUE TITLE
Fix TriBITS to find required Python version

### ISSUE THREAD

**Issue by bartlettroscoe** [TriBITSPub/TriBITS#184]
As part of CASL VERA work in:

* https://vminfo.casl.gov/trac/casl_phi_kanban/ticket/4849

it was found TriBITS no longer requires the version of Python that is required.  Instead, it just returns any old version of Python that it finds.  For that you case, you see:

```
-- Could NOT find PythonInterp: Found unsuitable version "2.6.6", but required is at least "2.7.9" (found /usr/bin/python)
-- PYTHON_EXECUTABLE='/usr/bin/python'
```

That version of Python is:

```
$ /usr/bin/python --version
Python 2.6.6
```

What the heck?

The handling of Python was updated in #58 to make it require the requested version of Python.  But it seems that behavior was broken at some point.  My guess is that this behavior was broken in #148.

@sethrj, have you seen this behavior of TriBITS for finding Python?

**Comment by bartlettroscoe** [comment]
This change appears to fix this:

```
diff --git a/tribits/core/package_arch/TribitsFindPythonInterp.cmake b/tribits/core/package_arch/TribitsFindPythonInterp.cmake
index 7a2dd7f..c725a1d 100644
--- a/tribits/core/package_arch/TribitsFindPythonInterp.cmake
+++ b/tribits/core/package_arch/TribitsFindPythonInterp.cmake
@@ -55,7 +55,12 @@ MACRO(TRIBITS_FIND_PYTHON)
   IF (TRIBITS_FIND_PYTHON_UNITTEST)
     SET(PYTHON_EXECUTABLE  ${PYTHON_EXECUTABLE_UNITTEST_VAL})
   ELSE()
-    FIND_PACKAGE(PythonInterp)
+    IF (${PROJECT_NAME}_REQUIRES_PYTHON)
+      SET(PythonInterp_REQUIRED_ARG "REQUIRED")
+    ELSE()
+      SET(PythonInterp_REQUIRED_ARG "")
+    ENDIF()
+    FIND_PACKAGE(PythonInterp ${PythonInterp_REQUIRED_ARG})
   ENDIF()
 ENDMACRO()
````

I will push to a branch and test with Trilinos as well (which does not require Python).

I also need to consider how to add an automated test for this behavior (since no existing test catches this obviously).

**Comment by bartlettroscoe** [comment]
This got fixed in commit 5dc14284e413394568d1e5b7b837d53778c2ae9b that I pushed to the casl-vera and github 'master' branches.  I put in the wrong GitHub issue ID #148 instead of #184 so it did not link into this issue. 

I tested this with CASL VERA and this works.  I am closing as complete.
